addons/project/models/project.py

```diff
@@ -1838,15 +1838,12 @@ def write(self, vals):
             tasks._populate_missing_personal_stages()
 
         # user_ids change: update date_assign
-        if not vals.get('user_ids', []) and 'user_ids' in vals:
-            self.date_assign = False
-        elif 'user_ids' in vals:
-            for element in self:
-                new_user_ids = element.user_ids
-                if len(new_user_ids) == 0:
-                    element.date_assign = False
+        if 'user_ids' in vals:
+            for task in self:
+                if len(task.user_ids) == 0:
```

Comment:
Equivalent to: 
```suggestion
                if not task.user_ids:
```
😉